python/hackathon_app/sheets.py

```diff
@@ -52,14 +56,18 @@ def get_hackathons(self) -> Sequence["Hackathon"]:
 
     def register_user(self, *, hackathon: str, user: "User"):
         """Register user to a hackathon"""
-        if not self.users.is_created(user):
-            self.users.create(user)
+        is_created = self.users.is_created(user)
         registrant = Registrant(
             user_email=user.email,
             hackathon_name=hackathon,
             date_registered=datetime.datetime.now(),
             attended=None,
         )
+        is_registered = self.registrations.is_registered(registrant)
+        if is_created and is_registered:
+            raise SheetError(f"User already exists and is registered to {hackathon}")
+        if not self.users.is_created(user):
+            self.users.create(user)
```

Comment:
can you update this per our conversation? (I think you already made the change locally, just need to push it up)